src/components/ChatEntry.js

```diff
@@ -1,22 +1,55 @@
 import React from 'react';
 import './ChatEntry.css';
+
 import PropTypes from 'prop-types';
+import TimeStamp from './TimeStamp';
+import { useState } from 'react';
 
 const ChatEntry = (props) => {
+
+  //deconstruct
+  const { sender, body, timeStamp, liked, redFilledHeartCount, setRedFilledHeartCount } = props;
+  //remember initial state liked = false
+  const [isEmptyHeart, setIsEmptyHeart] = useState(liked);
+
+  //function to toggle 
+  const updateHeart = () => {
+    setIsEmptyHeart(!isEmptyHeart);
+    
+    //checkpoint - if true
+    //increment
+    if (!isEmptyHeart) {
+      setRedFilledHeartCount(redFilledHeartCount + 1);
+    }
+    else {
+      // decrement
+      setRedFilledHeartCount(redFilledHeartCount - 1);
+    }
+
+ 
+  }
   return (
     <div className="chat-entry local">
-      <h2 className="entry-name">Replace with name of sender</h2>
+      <h2 className="entry-name">{sender}</h2>
       <section className="entry-bubble">
-        <p>Replace with body of ChatEntry</p>
-        <p className="entry-time">Replace with TimeStamp component</p>
-        <button className="like">🤍</button>
+        <p>{body}</p>
+        <p className="entry-time">
+          <TimeStamp time={timeStamp}></TimeStamp>
+        </p>
+        <button className="like" onClick={updateHeart}>
+          {isEmptyHeart ? '❤️' : '🤍'} 
+        </button>
       </section>
     </div>
   );
 };
 
 ChatEntry.propTypes = {
```

Comment:
Yay, prop types! It would be good to include all of the props that are being passed in, like `redFilledHeartCount` and `setRedFilledHeartCount`.